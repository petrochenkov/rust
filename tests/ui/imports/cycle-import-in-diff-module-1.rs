// similar `cycle-import-in-diff-module-0.rs`

mod a {
    pub(crate) use crate::s;
    //~^ ERROR unresolved imports `crate::s`, `self::a::s`
    //~| ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
}
mod b {
    pub mod s {}
}
use self::b::*;
use self::a::s; //~ ERROR cannot determine resolution for the import

fn main() {}
