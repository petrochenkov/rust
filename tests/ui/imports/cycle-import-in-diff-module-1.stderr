error: cannot determine resolution for the import
  --> $DIR/cycle-import-in-diff-module-1.rs:4:20
   |
LL |     pub(crate) use crate::s;
   |                    ^^^^^^^^

error: cannot determine resolution for the import
  --> $DIR/cycle-import-in-diff-module-1.rs:4:20
   |
LL |     pub(crate) use crate::s;
   |                    ^^^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/cycle-import-in-diff-module-1.rs:13:5
   |
LL | use self::a::s;
   |     ^^^^^^^^^^

error[E0432]: unresolved imports `crate::s`, `self::a::s`
  --> $DIR/cycle-import-in-diff-module-1.rs:4:20
   |
LL |     pub(crate) use crate::s;
   |                    ^^^^^^^^
...
LL | use self::a::s;
   |     ^^^^^^^^^^

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0432`.
